pkg/handler/v1/provider/openstack.go

```diff
@@ -21,37 +21,51 @@ import (
 	"crypto/x509"
 	"fmt"
 	"net/http"
+	"strings"
+
+	"k8c.io/kubermatic/v2/pkg/handler/middleware"
+	k8cerrors "k8c.io/kubermatic/v2/pkg/util/errors"
 
 	"github.com/go-kit/kit/endpoint"
 
 	providercommon "k8c.io/kubermatic/v2/pkg/handler/common/provider"
 	"k8c.io/kubermatic/v2/pkg/handler/v1/common"
 	"k8c.io/kubermatic/v2/pkg/provider"
+	"k8c.io/kubermatic/v2/pkg/resources"
 )
 
-type credentials struct {
-	username, password, tenant, tenantID, domain string
-}
-
-func getAuthInfo(ctx context.Context, req OpenstackReq, userInfoGetter provider.UserInfoGetter, presetsProvider provider.PresetProvider) (*provider.UserInfo, *credentials, error) {
-	var cred *credentials
+func getAuthInfo(ctx context.Context, req OpenstackReq, userInfoGetter provider.UserInfoGetter, presetsProvider provider.PresetProvider) (*provider.UserInfo, *resources.OpenstackCredentials, error) {
+	var cred *resources.OpenstackCredentials
 	userInfo, err := userInfoGetter(ctx, "")
 	if err != nil {
 		return nil, nil, common.KubernetesErrorToHTTPError(err)
 	}
+
+	t := ctx.Value(middleware.RawTokenContextKey)
+	token, ok := t.(string)
+	if !ok || token == "" {
+		return nil, nil, k8cerrors.NewNotAuthorized()
+	}
```

Comment:
The token is required, but we only use it if `OIDCAuthentication` is true. What if OIDCAuthentication is not used, do we need to return NotAuthorized if the token is missing?

Reply:
Resolved, the token is always present so no issue here.